File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/notifications/NotificationsAdapter.java
package pl.szczodrzynski.edziennik.ui.modules.notifications;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffColorFilter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import pl.szczodrzynski.edziennik.App;
import pl.szczodrzynski.edziennik.R;
import pl.szczodrzynski.edziennik.utils.models.Date;
import pl.szczodrzynski.edziennik.utils.models.Notification;

import static pl.szczodrzynski.edziennik.utils.Utils.d;

public class NotificationsAdapter extends RecyclerView.Adapter<NotificationsAdapter.ViewHolder> {
    private static final String TAG = "NotificationsAdapter";
    private Context context;
    private List<Notification> notificationList;

    //getting the context and product list with constructor
    public NotificationsAdapter(Context mCtx, List<Notification> notificationList) {
        this.context = mCtx;
        this.notificationList = notificationList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        //inflating and returning our view holder
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.row_notifications_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        App app = (App) context.getApplicationContext();

        Notification notification = notificationList.get(position);

        holder.notificationsItemDate.setText(Date.fromMillis(notification.addedDate).getFormattedString());
        holder.notificationsItemText.setText(notification.text);
        holder.notificationsItemTitle.setText(notification.title);
        holder.notificationsItemType.setText(Notification.stringType(context, notification.type));

        holder.notificationsItemCard.setOnClickListener((v -> {
            Intent intent = new Intent("android.intent.action.MAIN");
            notification.fillIntent(intent);

            d(TAG, "notification with item "+notification.redirectFragmentId+" extras "+(intent.getExtras() == null ? "null" : intent.getExtras().toString()));

            //Log.d(TAG, "Got date "+intent.getLongExtra("timetableDate", 0));

            if (notification.profileId != -1 && notification.profileId != app.profile.getId() && context instanceof Activity) {
                Toast.makeText(app, app.getString(R.string.toast_changing_profile), Toast.LENGTH_LONG).show();
            }
            app.sendBroadcast(intent);
        }));

        if (!notification.seen) {
            holder.notificationsItemText.setBackground(context.getResources().getDrawable(R.drawable.bg_rounded_8dp));
            holder.notificationsItemText.getBackground().setColorFilter(new PorterDuffColorFilter(0x692196f3, PorterDuff.Mode.MULTIPLY));
        }
        else {
            holder.notificationsItemText.setBackground(null);
        }

    }

    @Override
    public int getItemCount() {
        return notificationList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        CardView notificationsItemCard;
        TextView notificationsItemDate;
        TextView notificationsItemText;
        TextView notificationsItemTitle;
        TextView notificationsItemType;

        ViewHolder(View itemView) {
            super(itemView);
            notificationsItemCard = itemView.findViewById(R.id.notificationsItemCard);
            notificationsItemDate = itemView.findViewById(R.id.notificationsItemDate);
            notificationsItemText = itemView.findViewById(R.id.notificationsItemText);
            notificationsItemTitle = itemView.findViewById(R.id.notificationsItemTitle);
            notificationsItemType = itemView.findViewById(R.id.notificationsItemType);
        }
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/notifications/NotificationsAdapter.kt
package pl.szczodrzynski.edziennik.ui.modules.notifications

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import pl.szczodrzynski.edziennik.*
import pl.szczodrzynski.edziennik.data.db.modules.notification.Notification
import pl.szczodrzynski.edziennik.data.db.modules.notification.getNotificationTitle
import pl.szczodrzynski.edziennik.utils.Utils.d
import pl.szczodrzynski.edziennik.utils.models.Date

class NotificationsAdapter(
        private val context: Context
) : RecyclerView.Adapter<NotificationsAdapter.ViewHolder>() {
    companion object {
        private const val TAG = "NotificationsAdapter"
    }

    var items = listOf<Notification>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.row_notifications_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val app = context.applicationContext as App

        val notification = items[position]

        val date = Date.fromMillis(notification.addedDate).formattedString
        val colorSecondary = android.R.attr.textColorSecondary.resolveAttr(context)

        holder.title.text = notification.text
        holder.profileDate.text = listOf(
                notification.profileName ?: "",
                " • ",
                date.asColoredSpannable(colorSecondary)
        ).concat()
        holder.type.text = context.getNotificationTitle(notification.type)

        holder.root.onClick {
            val intent = Intent("android.intent.action.MAIN")
            notification.fillIntent(intent)

            d(TAG, "notification with item " + notification.viewId + " extras " + if (intent.extras == null) "null" else intent.extras!!.toString())

            //Log.d(TAG, "Got date "+intent.getLongExtra("timetableDate", 0));

            if (notification.profileId != -1 && notification.profileId != app.profile.id && context is Activity) {
                Toast.makeText(app, app.getString(R.string.toast_changing_profile), Toast.LENGTH_LONG).show()
            }
            app.sendBroadcast(intent)
        }
    }

    override fun getItemCount() = items.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var root = itemView
        var title: TextView = itemView.findViewById(R.id.title)
        var profileDate: TextView = itemView.findViewById(R.id.profileDate)
        var type: TextView = itemView.findViewById(R.id.type)
    }
}
